### PR TITLE
fix: PopoverTriggerChildProps should be exported

### PR DESCRIPTION
**Pull request by ling1726**
PopoverTriggerChildProps was not exported. Without this type it's not easy to now what props are being returned by the `PopoverTrigger` if users want to implement their own custom trigger or use a render function to do so

Also fixes the custom trigger story for Popover to actually create a custom trigger

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `84.575 kB`<br />`21.195 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.874 kB`<br />`13.864 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `14.996 kB`<br />`6.013 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `68.842 kB`<br />`19.205 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.827 kB`<br />`17.599 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.273 kB`<br />`52.499 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                                 |   `53.992 kB`<br />`15.25 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `102.802 kB`<br />`31.516 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/8954cd0e856c4fdda7745f40d3c30916a6e24f6b'>8954cd0e856c4fdda7745f40d3c30916a6e24f6b</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-components/src/index.ts [hunk 1]
```diff
@@ -471,6 +471,7 @@ export type {
   PopoverSurfaceState,
   PopoverTriggerProps,
   PopoverTriggerState,
+  PopoverTriggerChildProps,
```
Comment (Hotell) [inline packages/react-components/react-components/src/index.ts:474]: can you provide more context why it should be exported ?

**Review by Hotell — APPROVED** [review]

File: packages/react-components/react-components/src/index.ts (hunk 1 above)
Comment (ling1726) [reply packages/react-components/react-components/src/index.ts:474]: Updated the PR description

Comment (khmakoto) [inline packages/react-components/react-components/src/index.ts:474]: _nit:_ Keep these ordered alphabetically

```suggestion
  PopoverTriggerChildProps,
  PopoverTriggerProps,
  PopoverTriggerState,
```

**Review by khmakoto — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Review by bsunderhus — APPROVED** [review]

**Pull request merged**